### ISSUE TITLE
Odoc: valid module name clashes cause issues

### ISSUE THREAD

**Issue by jonludlam** [ocaml/dune#1645]

**Comment by diml** [comment]
Hmm yh, I believe it's because odoc works with packages rather than libraries. I don't know how we can fix this in Dune :/

**Comment by jonludlam** [comment]
I've updated the description to make it clear that this is for sub-libraries (repro.one and repro.two)

**Comment by jonludlam** [comment]
This is related to odoc#148

**Comment by avsm** [comment]
I'm still seeing this in ocamllabs/ocaml-ctypes#588 with odoc 1.4.0, need to investigate more:


```
 $ dune build @doc
Error: Conflicting implementations for virtual library "ctypes-foreign":
- "ctypes-foreign.threaded" in _build/default/src/ctypes-foreign-threaded
- "ctypes-foreign.unthreaded" in _build/default/src/ctypes-foreign-unthreaded
Multiple rules generated for _build/default/_doc/_html/ctypes-foreign/Foreign/index.html:
- <internal location>
- <internal location>
```

**Comment by rgrinberg** [comment]
@jonludlam is there a corresponding odoc bug? I don't think this can be fixed until odoc addresses this.

**Comment by jonludlam** [comment]
@avsm which version of dune are you using? This ought to be addressed by https://github.com/ocaml/dune/pull/2141

@rgrinberg there's no issue on the odoc repo yet. I'll make one when I've got a moment.

**Comment by avsm** [comment]
@jonludlam twas with the released dune 1.9.3 -- I'll try with the master branch shortly

**Comment by davesnx** [comment]
Facing this issue and found https://github.com/ocaml/odoc/issues/148 has been resolved. Is there any chance to relook into this?

**Comment by Khady** [comment]
Is it correct to say that this is (at least partially) fixed by the `@doc-new` rule?

**Comment by emillon** [comment]
ping @jonludlam for this question
(though I wouldn't close issues fixed by doc-new until we know how these new rules are going to be enabled)

**Comment by jonludlam** [comment]
Yes, `doc-new` should not have this problem. See the test cases for [doc rules](https://github.com/ocaml/dune/blob/main/test/blackbox-tests/test-cases/odoc/github1645.t) and [doc-new rules](https://github.com/ocaml/dune/blob/main/test/blackbox-tests/test-cases/odoc/new/github1645.t)
